spring-amqp/src/main/java/org/springframework/amqp/core/Address.java

```diff
@@ -58,6 +60,10 @@ public Address(String address) {
 			this.exchangeName = "";
 			this.routingKey = "";
 		}
+		else if (address.startsWith(AMQ_RABBITMQ_REPLY_TO)) {
+			this.routingKey = address;
+			this.exchangeName = "";
```

Comment:
H-m-m. It is strange that we have missed this fix for the first implementation...
It isn't breaking change, unless the movement of that `AMQ_RABBITMQ_REPLY_TO` const.
Will hope it won't harm any upgraded app to the AMQP 1.4.2
Merging and cherry-picking...